ksql-engine/src/test/resources/query-validation-tests/key-field.json

```diff
@@ -1156,19 +1315,27 @@
         {"topic": "OUTPUT", "key":"2|+|1", "value": {"FOO": 1, "BAR": 2, "KSQL_COL_2": 1}}
       ],
       "post": {
-        "issues": ["key field of output is INCORRECT: should be null - (see Issue #2 above)"],
         "sources": [
-          {"name": "OUTPUT", "type": "table", "keyField": {"name": "KSQL_INTERNAL_COL_1|+|KSQL_INTERNAL_COL_0", "schema": {"type": "STRING"}}}
+          {"name": "OUTPUT", "type": "table", "keyField": {"name": null, "legacyName": "KSQL_INTERNAL_COL_1|+|KSQL_INTERNAL_COL_0", "legacySchema": {"type": "STRING"}}}
```

Comment:
Fascinating... will we change this going forward when we have structured keys? I would expect the key to be `(bar, foo)`

Reply:
Not sure at this stage.  Likely, initially, we'll support it where a single field within the value contains the key. So in this example there would need to be a field that was a STRUCT with two fields 'bar' and 'foo'.  Of course, that field could be nested within some other struct.

Supporting the key spread across multiple fields in the value, as you propose, is likely more tricky to implement, but we'll see when I get to that point.